tcptracer-bpf.c

```diff
@@ -18,98 +22,67 @@
-	.namespace = "",
-};
-
-/* This is a key/value store with the keys being an ipv6_tuple_t for send & recv calls
- * and the values being the struct conn_stats_t *.
- */
-struct bpf_map_def SEC("maps/tcp_stats_ipv6") tcp_stats_ipv6 = {
-	.type = BPF_MAP_TYPE_HASH,
-	.key_size = sizeof(struct ipv6_tuple_t),
-	.value_size = sizeof(struct conn_stats_t),
-	.max_entries = 32768,
-	.pinning = 0,
-	.namespace = "",
+#define MAX_MSG_SIZE 1024
+
+
+// struct addr_info_t
+// {
+// 	struct sockaddr *addr;
+// 	size_t *addrlen;
+// };
+
+// #define MAX_MSG_SIZE 1024
+// BPF_PERF_OUTPUT(syscall_write_events);
+
+// // This needs to match exactly with the Go version of the struct.
+// struct syscall_write_event_t
+// {
+// 	// We split attributes into a separate struct, because BPF gets upset if you do lots of
+// 	// size arithmetic. This makes it so that it's attributes followed by message.
+// 	struct attr_t
+// 	{
+// 		int event_type;
+// 		int fd;
+// 		int bytes;
+// 		// Care needs to be taken as only msg_size bytes of msg are guaranteed
+// 		// to be valid.
+// 		int msg_size;
+// 	} attr;
+// 	char msg[MAX_MSG_SIZE];
+// };
+
+// const int kEventTypeSyscallAddrEvent = 1;
+// const int kEventTypeSyscallWriteEvent = 2;
+// const int kEventTypeSyscallCloseEvent = 3;
+
+// BPF programs are limited to a 512-byte stack. We store this value per CPU
+// and use it as a heap allocated value.
+
+struct bpf_map_def SEC("maps/write_buffer_heap") write_buffer_heap = {
+        .type = BPF_MAP_TYPE_PERCPU_ARRAY,
+        .key_size = sizeof(int),
+        .value_size = sizeof(char[MAX_MSG_SIZE]),
+        .max_entries = 1,
+        .pinning = 0,
+        .namespace = "",
 };
 
-/* These maps are used to match the kprobe & kretprobe of connect for IPv4 */
-/* This is a key/value store with the keys being a pid
- * and the values being a struct sock *.
- */
-struct bpf_map_def SEC("maps/connectsock_ipv4") connectsock_ipv4 = {
-	.type = BPF_MAP_TYPE_HASH,
-	.key_size = sizeof(__u64),
-	.value_size = sizeof(void *),
-	.max_entries = 1024,
-	.pinning = 0,
-	.namespace = "",
-};
-
-/* These maps are used to match the kprobe & kretprobe of connect for IPv6 */
-/* This is a key/value store with the keys being a pid
- * and the values being a struct sock *.
- */
-struct bpf_map_def SEC("maps/connectsock_ipv6") connectsock_ipv6 = {
-	.type = BPF_MAP_TYPE_HASH,
-	.key_size = sizeof(__u64),
-	.value_size = sizeof(void *),
-	.max_entries = 1024,
-	.pinning = 0,
-	.namespace = "",
-};
-
-/* This map is used to match the kprobe & kretprobe of udp_recvmsg */
-/* This is a key/value store with the keys being a pid
- * and the values being a struct sock *.
- */
-struct bpf_map_def SEC("maps/udp_recv_sock") udp_recv_sock = {
-	.type = BPF_MAP_TYPE_HASH,
-	.key_size = sizeof(__u64),
-	.value_size = sizeof(void *),
-	.max_entries = 1024,
-	.pinning = 0,
-	.namespace = "",
+// The set of file descriptors we are tracking.
+struct bpf_map_def SEC("maps/active_fds") active_fds = {
```

Comment:
All maps have been moved to a separate `.h` file except these 2 new maps. Seems like they should go as well.

Reply:
done